common/src/primitives/per_thousand.rs

```diff
@@ -78,6 +93,16 @@ impl Decode for PerThousand {
     }
 }
 
+impl Display for PerThousand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}%",
+            Amount::from_atoms(self.0.into()).into_fixedpoint_str(1)
```

Comment:
Why did you remove the into_percentage_str() function? Why not call it in Display?

Reply:
I've added `into_percentage_str` back.
Also, I've decided not to call it in `Display`, but instead implement them both via a separate function
```
fn write_as_percentage_str(&self, dest: &mut impl std::fmt::Write) -> std::fmt::Result {
    write!(
        dest,
        "{}%",
        Amount::from_atoms(self.0.into()).into_fixedpoint_str(1)
    )
}
```
This may look kind of redundant, but it's also a bit nicer, because `Display` doesn't have to do an unneeded string allocation.